libgnucash/engine/gnc-commodity.cpp

```diff
@@ -191,51 +191,72 @@ static QuoteSourceList single_quote_sources =
+    { false, SOURCE_SINGLE, "comdirect, DE", "comdirect" },
+    { false, SOURCE_SINGLE, "Consors Bank, DE", "consorsbank" },
     { false, SOURCE_SINGLE, "Deka Investments, DE", "deka" },
-    { false, SOURCE_SINGLE, "Dutch", "dutch" },
     { false, SOURCE_SINGLE, "DWS, DE", "dwsfunds" },
     { false, SOURCE_SINGLE, "Financial Times Funds service, GB", "ftfunds" },
     { false, SOURCE_SINGLE, "Finanzpartner, DE", "finanzpartner" },
-    { false, SOURCE_SINGLE, "GoldMoney spot rates, JE", "goldmoney" },
+    { false, SOURCE_SINGLE, "FondsWeb, DE", "fondsweb" },
+    { false, SOURCE_SINGLE, "GoldMoney precious metals", "goldmoney" },
     { false, SOURCE_SINGLE, "Google Web, US Stocks", "googleweb" },
-    { false, SOURCE_SINGLE, "India Mutual", "indiamutual" },
+    { false, SOURCE_SINGLE, "IEX (Investors Exchange), US", "iexcloud" },
+    { false, SOURCE_SINGLE, "Market Watch", "marketwatch" },
+    { false, SOURCE_SINGLE, "Morningstar, AU", "morningstarau" },
+    { false, SOURCE_SINGLE, "Morningstar, CH", "morningstarch" },
     { false, SOURCE_SINGLE, "Morningstar, GB", "morningstaruk" },
     { false, SOURCE_SINGLE, "Morningstar, JP", "morningstarjp" },
+    { false, SOURCE_SINGLE, "Motley Fool", "fool" },
     { false, SOURCE_SINGLE, "New Zealand stock eXchange, NZ", "nzx" },
+    { false, SOURCE_SINGLE, "NSE (National Stock Exchange), IN", "nseindia" },
+    { false, SOURCE_SINGLE, "OnVista", "onvista"},
     { false, SOURCE_SINGLE, "Paris Stock Exchange/Boursorama, FR", "bourso" },
-    { false, SOURCE_SINGLE, "Romania", "romania" },
+    { false, SOURCE_SINGLE, "S-Investor", "sinvestor"},
+    { false, SOURCE_SINGLE, "Sharenet, ZA", "za" },
     { false, SOURCE_SINGLE, "SIX Swiss Exchange shares, CH", "six" },
     { false, SOURCE_SINGLE, "Skandinaviska Enskilda Banken, SE", "seb_funds" },
-    { false, SOURCE_SINGLE, "Sharenet, ZA", "za" },
+    { false, SOURCE_SINGLE, "StockData", "stockdata" },
+    { false, SOURCE_SINGLE, "Stooq, PL", "stooq" },
+    { false, SOURCE_SINGLE, "T. Rowe Price, US", "troweprice" },
+    { false, SOURCE_SINGLE, "Tesouro Direto bonds, BR", "tesouro_direto" },
     { false, SOURCE_SINGLE, "TIAA-CREF, US", "tiaacref" },
     { false, SOURCE_SINGLE, "Toronto Stock eXchange, CA", "tsx" },
-    { false, SOURCE_SINGLE, "T. Rowe Price", "troweprice" },
-    { false, SOURCE_SINGLE, "T. Rowe Price, US", "troweprice_direct" },
+    { false, SOURCE_SINGLE, "Tradegate, DE", "tradegate" },
+    { false, SOURCE_SINGLE, "Treasury Direct bonds, US", "treasurydirect" },
+    { false, SOURCE_SINGLE, "Twelve Data", "twelvedata" },
     { false, SOURCE_SINGLE, "Union Investment, DE", "unionfunds" },
     { false, SOURCE_SINGLE, "US Govt. Thrift Savings Plan", "tsp" },
+    { false, SOURCE_SINGLE, "XETRA, DE", "xetra" },
     { false, SOURCE_SINGLE, "Yahoo as JSON", "yahoo_json" },
     { false, SOURCE_SINGLE, "Yahoo Web", "yahooweb" },
     { false, SOURCE_SINGLE, "YH Finance (FinanceAPI)", "financeapi" },
```

Comment:
@fellen I think YH Financial is just wrapping Yahoo JSON. Their URLs are exactly the same as Yahoo!'s except for the site (yfapi.net instead of query2.finance.yahoo.com).

I don't think that it matters where a source is headquartered any more than its TLD does. @pkryger suggested `global` and I suggested `multinational`. Maybe we're trying to cram too much meaning into the list items and should be pointing users at the docs/wiki instead?

Separate issue, shouldn't the names be marked for translation?

Reply:
Given how much discussion it causes, I begin to think that leaving region out for sources that cover multiple regions/countries/asset classes is the way to go. 

I'd rather apply a relatively simple rule how to mark such sources, with a short - yet descriptive - name. To my (non-native, non-US based) brain `multinational` seems the best description. The issue I have with this term it's its length. I think that was one of the reasons I suggested `global`. I think I like `US+` (again, biased by my personal experience), but perhaps it is not that obvious. How about yet another alternative: `US, ...`?

But then we're getting into semantics. It seems that both `twelvedata` and `stockdata` are non-US based companies. [The former is Singapore](https://tracxn.com/d/companies/twelve-data/__AIoSizGMSOpZLV9-Yb1Qr450ru1tmshFw1B1-XtVRWo), while the latter is UK (guessing from their choice of "Governing Laws" and "Dispute Resolution" in their [TOS](https://www.stockdata.org/tos). I have been unable to find to find their registration in Companies House or their physical address. So marking them as US (which they cover - I guess primarily!) seems odd to me. I think I subconsciously put them under one umbrella with `marketwatch`, `bloomberg`, and `yahoo` (all variants, `financeapi` included).

As a counter example there is `stooq`. I guess it is a goto for Polish based stock, funds, etc.. And it's a Warsaw, Poland based company and the source is marked with `PL` region. OTOH, it provides quotes for stock traded in US, UK, Germany, and Japan. Perhaps there are others that do provide similar coverage.

Let me know what you think. And which resolution (and name) sounds best for you. I think I was mostly voicing my opinions here, and  happy to use whatever you choose.

> Separate issue, shouldn't the names be marked for translation?

This seems like a non trivial issue to me, starting with what do to with sources like `bvb` or `borsa_italiana`: use english version of their names, even though some of them seem to me like proper nouns? What should be source for these translations to English?

Reply:
MarketWatch is a Dow Jones property along with Barron's and The Wall Street Journal. Dow Jones is in turn a property of the Murdoch family's News Corp. Very much a direct competitor with Bloomberg minus the fancy hardware.

Reply:
> Given how much discussion it causes, I begin to think that leaving region out for sources that cover multiple regions/countries/asset classes is the way to go.
…
As a counter example there is stooq. I guess it is a goto for Polish based stock, funds, etc.. And it's a Warsaw, Poland based company and the source is marked with PL region. OTOH, it provides quotes for stock traded in US, UK, Germany, and Japan. Perhaps there are others that do provide similar coverage.

Which is why I think it's misleading to assign country codes on the menu at all.

Reply:
> But then we're getting into semantics. It seems that both `twelvedata` and `stockdata` are non-US based companies. [](https://tracxn.com/d/companies/twelve-data/__AIoSizGMSOpZLV9-Yb1Qr450ru1tmshFw1B1-XtVRWo)

That can be important. Imagine a future president Trump will build a Great Firewall disconnecting the US from the internet. Then they will be the only remaining sources. ;-)

Reply:
Then SG+, PL+, … would be helpful.

When I searched symbols of german funds, i had started to add the region and came down to ~5 german funds companies, which at that time all used german WKNs as symbol. On that way I had to go through all sources because their names were not very meaningful to me. Later users can now restrict their search.

One side effect: the EU has better privacy protection than many other countries, which might store cookies without query, if you visit a there registered site.

Reply:
Moving discussion to the main thread